InterfaceParser/parsers/rpc_base.py

```diff
@@ -0,0 +1,782 @@
+            elif subelement.tag == "warning":
+                warnings.append(self._parse_simple_element(subelement))
+            else:
+                subelements.append(subelement)
+
+        params["description"] = description
+        params["design_description"] = design_description
+        params["issues"] = issues
+        params["todos"] = todos
+        params["history"] = history
+
+        return params, subelements, attrib
+
+    @staticmethod
+    def _parse_simple_element(element):
+        """Parse element as simple element and returns it's text.
+
+        Element is simple when it contains no subelements and attributes.
+
+        :param element: an instance of Element (from ElementTree)
+        :return: element text if present or empty string if not
+        """
+        if len(element) != 0:
+            raise ParseError("Unexpected subelements in '{}'".format(element.tag))
+        if len(element.attrib) != 0:
+            raise ParseError("Unexpected attributes in '{}'".format(element.tag))
+        return element.text if element.text is not None else ""
+
+    @staticmethod
+    def _parse_issue(element):
+        """Parse element as issue.
+
+        Issue must not contain subelements and attributes.
+
+        :param element: an instance of Element (from ElementTree)
+        :return: an instance of model.Issue
+        """
+        if len(element) != 0:
+            raise ParseError("Unexpected subelements in issue")
+        if "creator" not in element.attrib:
+            raise ParseError("No creator in issue")
+        if len(element.attrib) != 1:
+            raise ParseError("Unexpected attributes in issue")
+
+        return Issue(
+            creator=element.attrib["creator"],
+            value=element.text if element.text is not None else "")
+
+    def _parse_enum_element(self, element):
+        """Parse element as element of enumeration.
+
+        :param element: an instance of Element (from ElementTree)
+        :return: an instance of model.EnumElement
+        """
+        params, subelements, attrib = self._parse_base_item(element, "")
+
+        if len(subelements) != 0:
+            raise ParseError("Unexpected subelements in enum element")
+
+        self._ignore_attribute(attrib, "scope")
+        self._ignore_attribute(attrib, "rootscreen")
+
+        for attribute in attrib:
+            if attribute == "value":
+                try:
+                    params[attribute] = int(attrib[attribute])
+                except:
+                    raise ParseError("Invalid value for enum element: '{}'".format(attrib[attribute]))
+            elif attribute in ["internal_name", "hexvalue", "deprecated", "removed"]:
+                params[attribute] = attrib[attribute]
+            elif attribute in ["since", "until"]:
+                params[attribute] = self._parse_version(attrib[attribute])
+
+        return EnumElement(**params)
+
+    def _parse_param(self, element, prefix):
```

Comment:
Can we rename this to `_parse_struct_param`?

Reply:
renamed